controllers/openstackdataplanerole_controller.go

```diff
@@ -216,21 +218,72 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
 			return result, nil
 		}
 
-		r.Log.Info("Set DataPlaneRoleReadyCondition true")
-		instance.Status.Conditions.Set(condition.FalseCondition(dataplanev1beta1.DataPlaneRoleReadyCondition, condition.InitReason, condition.SeverityInfo, dataplanev1beta1.DataPlaneRoleReadyWaitingMessage))
-	}
+		// Call DeployNovaExternalCompute individually for each node
+		var novaExternalCompute *novav1beta1.NovaExternalCompute
+		var novaReadyConditionsTrue []*condition.Condition
+		var novaErrors []error
+		for _, node := range nodes.Items {
+			nodeConfigMapName := fmt.Sprintf("dataplanenode-%s-inventory", node.Name)
+			result, novaExternalCompute, err = deployment.DeployNovaExternalCompute(
+				ctx,
+				helper,
+				&node,
+				ansibleSSHPrivateKeySecret,
+				nodeConfigMapName,
+				&instance.Status,
+				instance.Spec.NetworkAttachments,
+				instance.Spec.OpenStackAnsibleEERunnerImage)
+			if err != nil {
+				novaErrors = append(novaErrors, err)
+			}
+			novaReadyCondition := novaExternalCompute.Status.Conditions.Get(condition.ReadyCondition)
+			if novaExternalCompute.IsReady() {
+				r.Log.Info(fmt.Sprintf("NovaExternalCompute %s, IsReady true", node.Name))
+				novaReadyConditionsTrue = append(novaReadyConditionsTrue, novaReadyCondition)
+
+			}
+		}
+
+		// When any errors are found, wrap all into a single error, and return
+		// it
+		errStr := "DeployNovaExternalCompute error:"
+		if len(novaErrors) > 0 {
+			for _, err := range novaErrors {
+				errStr = fmt.Sprintf("%s: %s", errStr, err.Error())
+			}
+			err = errors.New(errStr)
+			return result, err
+		}
+
+		// Return when any condition is not ready, otherwise set the role as
+		// deployed.
+		if len(novaReadyConditionsTrue) < len(nodes.Items) {
+			r.Log.Info("Not all NovaExternalCompute ReadyConditions are true")
+			return result, nil
+		}
+
+		r.Log.Info("All NovaExternalCompute ReadyConditions are true")
+		instance.Status.Conditions.Set(condition.TrueCondition(dataplanev1beta1.NovaComputeReadyCondition, dataplanev1beta1.NovaComputeReadyMessage))
+		instance.Status.Deployed = true
+		r.Log.Info("Set ReadyCondition true")
+		instance.Status.Conditions.Set(condition.TrueCondition(condition.ReadyCondition, dataplanev1beta1.DataPlaneRoleReadyMessage))
+
+		// Explicitly set instance.Spec.Deploy = false
+		// We don't want another deploy triggered by any reconcile request, it should
+		// only be triggered when the user (or another controller) specifically
+		// sets it to true.
+		r.Log.Info("Set DeployStrategy.Deploy to false")
+		instance.Spec.DeployStrategy.Deploy = false
```

Comment:
resetting this back to False, after a user requested a deploy could be confusing